L1Trigger/L1TTrackMatch/plugins/L1TrackSelectionProducer.cc

```diff
@@ -294,6 +295,66 @@ class L1TrackSelectionProducer : public edm::global::EDProducer<> {
     std::vector<double> deltaZMax_;
   };
 
+  struct NNTrackSelector {
+    NNTrackSelector(tensorflow::Session* AssociationSesh,
+                    const double AssociationThreshold,
+                    const std::vector<double>& AssociationNetworkZ0binning,
+                    const double AssociationNetworkMaxPt,
+                    const std::vector<double>& AssociationNetworkEtaBounds,
+                    const std::vector<double>& AssociationNetworkZ0ResBins)
+        : AssociationSesh_(AssociationSesh),
+          AssociationThreshold_(AssociationThreshold),
+          z0_binning_(AssociationNetworkZ0binning),
+          max_pt_(AssociationNetworkMaxPt),
+          eta_bins_(AssociationNetworkEtaBounds),
+          res_bins_(AssociationNetworkZ0ResBins) {}
+
+    template <typename VertexType>
+    bool associate(const L1Track& t, const VertexType& v) const {
+      tensorflow::Tensor inputAssoc(tensorflow::DT_FLOAT, {1, 4});
+      std::vector<tensorflow::Tensor> outputAssoc;
+
+      float etaBit = t.getTanlBits() < pow(2, (int)(TTTrack_TrackWord::TrackBitWidths::kTanlSize - 1))
```

Comment:
for [pTBit](https://github.com/BenjaminRS/cmssw/pull/4/files#diff-e50984fabbf6fafabef0cd83b6e7fdd75fb2400b3925cfcf03d9a1752fd15de9R333) you use `<=` ... just making sure that using `<` here is correct.

Reply:
This is correct, the representations of pt and eta are slightly different. In the future I would like to refine this and use the ap_fixed datatypes directly